core/server/web/api/canary/admin/routes.js

```diff
@@ -67,6 +67,7 @@ module.exports = function apiRoutes() {
 
     // ## Users
     router.get('/users', mw.authAdminApi, http(apiCanary.users.browse));
+    router.get('/users/reset-all-passwords', mw.authAdminApi, http(apiCanary.users.resetAllPasswords));
```

Comment:
I don't think we use kebab casing anywhere else to name an endpoint :thinking:  It's usually underscore separated like: `/email_preview/posts/:id`,  `/members/stats/gross_volume`.

Also not sure about the name. What other names have you considered when picking this one?

Reply:
This should also **definitely not** be a GET route

Reply:
It's updated to `PUT /users/reset_all_passwords`

Reply:
Regarding the name, this is how we're talking about the feature in the designs so it feels like the most descriptive name, I'm open to suggestions though.

![image](https://user-images.githubusercontent.com/494686/120668278-9a6a7400-c48e-11eb-94a3-165a40bb4ce5.png)

Reply:
Imo "all" part is not necessary for the non-user facing naming. Naming it `reset_password` leaves a door open to do things like passing a list of emails that have been compromised to this endpoint and doing a "bulk reset password". Also, worth giving a thought if we want to thing of a "password" as a separate resource that happens to belong to a user or leave it as is :thinking:

Reply:
@tpatel to add to above comment I'm not sure the `PUT` is a correct method here either. I struggled to find a good reference. The best example we use often is Stipe's API. In [similar "action" situation](https://stripe.com/docs/api/setup_intents) they use `POST`:
```
  POST /v1/setup_intents/:id/confirm
  POST /v1/setup_intents/:id/cancel
```

What outside internal references did you check when deciding on the naming?

Reply:
I had started by looking at the internal usage of `POST`/`PUT` in Ghost to mimic the choices made by the team:

```
PUT /users/password => changePassword
PUT /authentication/passwordreset => resetPassword
POST /authentication/passwordreset => generateResetToken
```

As we're modifying the state of users, sending a password reset email AND deleting sessions, it's hard to know which one fits best and I arbitrarily chose `PUT`.

Now, reading [RFC 2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9) seems to point towards using `POST` instead. The main reason is that `PUT` is mentioned as being idempotent vs `POST` not having to be idempotent.

I've updated it ✅

Reply:
Yup. Makes sense :+1: